### PR TITLE
Add service to send any command to tahoma.

### PR DESCRIPTION
**Pull request by vlebourl**

### CODE REVIEW

File: custom_components/tahoma/__init__.py [hunk 1]
```diff
@@ -129,6 +135,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
+        """Handle execute command service."""
+        entity_registry = await hass.helpers.entity_registry.async_get_registry()
+        entity = entity_registry.entities.get(call.data.get("entity_id"))
+        await tahoma_coordinator.client.execute_command(
+            entity.unique_id,
+            Command(call.data.get("command"), call.data.get("args")),
+            "Home Assistant Service",
+        )
+
+    SCHEMA = vol.Schema(
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:148]: Can we rename SCHEMA to include the SERVICE it is for? Or include it inline in `async_register`.

File: custom_components/tahoma/__init__.py [hunk 2]
```diff
@@ -129,6 +135,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
         )
 
+    async def handle_execute_command(call):
+        """Handle execute command service."""
+        entity_registry = await hass.helpers.entity_registry.async_get_registry()
+        entity = entity_registry.entities.get(call.data.get("entity_id"))
+        await tahoma_coordinator.client.execute_command(
+            entity.unique_id,
+            Command(call.data.get("command"), call.data.get("args")),
+            "Home Assistant Service",
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:145]: Why not just call the execute_command method we already have in place?

File: custom_components/tahoma/__init__.py [hunk 3]
```diff
@@ -129,6 +135,29 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
+        {
+            vol.Required("entity_id"): cv.string,
+            vol.Required("command"): cv.string,
+            vol.Optional("args", default=[]): vol.All(
+                cv.ensure_list, [vol.Any(str, int)]
+            ),
+        }
+    )
+    hass.services.async_register(
+        DOMAIN, "execute_command", handle_execute_command, SCHEMA
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:158]: Please make a const for `"execute_command"` like the other services.

File: custom_components/tahoma/__init__.py (hunk 1 above)
Comment (vlebourl) [reply custom_components/tahoma/__init__.py:148]: sure!

File: custom_components/tahoma/__init__.py (hunk 2 above)
Comment (vlebourl) [reply custom_components/tahoma/__init__.py:145]: cause `AttributeError: 'RegistryEntry' object has no attribute 'async_execute_command'`

File: custom_components/tahoma/__init__.py (hunk 3 above)
Comment (vlebourl) [reply custom_components/tahoma/__init__.py]: done

**Review by tetienne — APPROVED** [review]

File: custom_components/tahoma/__init__.py [hunk 4]
```diff
@@ -137,6 +150,37 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
...
+                vol.Required("entity_id"): cv.string,
+                vol.Required("command"): cv.string,
+                vol.Optional("args", default=[]): vol.All(
+                    cv.ensure_list, [vol.Any(str, int)]
+                ),
+            }
+        ),
+    )
+
+    async def async_close_client(self, *_):
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:178]: Where is this change coming from? We should **not** close the client anymore.

**Review by iMicknl — CHANGES_REQUESTED** [review]
Please remove unrelated parts, like closing the client.

File: custom_components/tahoma/__init__.py (hunk 4 above)
Comment (vlebourl) [reply custom_components/tahoma/__init__.py:178]: leftovers from a rebase/merge I guess

File: custom_components/tahoma/__init__.py [hunk 5]
```diff
@@ -3,18 +3,29 @@
...
 
 from homeassistant import config_entries
 from homeassistant.components.scene import DOMAIN as SCENE
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_EXCLUDE, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.const import (
+    CONF_EXCLUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
```
Comment (iMicknl) [inline custom_components/tahoma/__init__.py:21]: Is `EVENT_HOMEASSISTANT_STOP` still used somewhere?

Comment (vlebourl) [reply custom_components/tahoma/__init__.py:21]: nope...

**Review by iMicknl — APPROVED** [review]

**Pull request merged**